lib/evse_security/crypto/openssl/openssl_provider.cpp

```diff
@@ -23,53 +23,72 @@ typedef struct ossl_lib_ctx_st OSSL_LIB_CTX;
 
 namespace evse_security {
 
-bool is_tpm_key_string(const std::string& private_key_pem) {
-    return private_key_pem.find("-----BEGIN TSS2 PRIVATE KEY-----") != std::string::npos;
+static auto KEY_HEADER_DEFAULT = "-----BEGIN PRIVATE KEY-----";
+static auto KEY_HEADER_TPM2 = "-----BEGIN TSS2 PRIVATE KEY-----";
+
+bool is_custom_key_string(const std::string& private_key_pem) {
```

Comment:
A think a comment is needed to explain that the function will return true for non-private key files (and non-PEM files)

Reply:
Makes sense, even changed the name to 'is_custom_private_key'. Will also make a PR for libocpp related to this, since there this function is being used.